vowpalwabbit/global_data.cc

```diff
@@ -257,14 +258,7 @@ VW_WARNING_DISABLE_DEPRECATED_USAGE
 
 vw::vw() : options(nullptr, nullptr)
 {
-  sd = &calloc_or_throw<shared_data>();
-  sd->dump_interval = 1.;  // next update progress dump
-  sd->contraction = 1.;
-  sd->first_observed_label = FLT_MAX;
-  sd->is_more_than_two_labels_observed = false;
-  sd->max_label = 0;
-  sd->min_label = 0;
-
+  sd = new shared_data();
```

Comment:
could this be made a unique_ptr?

Reply:
A seeded copy of the vw struct may point to an existing shared_data object. Because of that a shared_ptr is probably required to express the ownership. I'm open to making it a shared_ptr but think it should be a separate pr